### PR TITLE
Don't change file encoding unnecessarily.

### LINKED ISSUE #1908: Unicode error can occur when viewing diff for PR files
1. Checkout a different branch than PR #1888 as an example
2. Right click and "View Changes" of any `zh-CN.resx` file.
3. Observe error

**_Note:_** If you are actually on the branch this error does not occur.

### Images

image

image

### PR DESCRIPTION
**Pull request by grokys**
When comparing files in a PR. we're currently taking a look at the file on disk and converting the encoding of the LHS of the comparison to match the on-disk file. However, if the file doesn't exist locally, we were returning `Encoding.Default` for this value, which was causing the unicode errors seen in #1908.

This PR changes to behavior to return `null` if the file doesn't exist locally, and if a `null` encoding is passed to `PullRequestService.ExtractToTempFile` then just write the contents out without any encoding.

### Related

- #1010: Checked out PR file diffs sometimes complain about different encodings

Fixes #1908.

### CODE REVIEW

**Comment by meaghanlewis** [comment]
This LGTM 👍

File: src/GitHub.App/Services/PullRequestService.cs [hunk 1]
```diff
@@ -769,7 +769,7 @@ public Encoding GetEncoding(ILocalRepositoryModel repository, string relativePat
                 }
             }
 
-            return Encoding.Default;
+            return null;
```
Comment (jcansdale) [inline src/GitHub.App/Services/PullRequestService.cs]: I believe can we can simply replace this with the following and avoid having to spacial case `null`.

```suggestion
            return new UTF8Encoding(false);
```

This makes the `GetEncoding` responsible for returning `Encoding.UTF8` if the target file has a preamble or `new UTF8Encoding(false)` if the target file doesn't exist or have a preamble. UTF8 without preamble is Git's default text encoding.

It's possible we could always use `new UTF8Encoding(false)` and avoid this issue completely. The question is what a diff looks like between UTF8 files with and without preamble. I didn't know `new UTF8Encoding(false)` was a thing until today.

Comment (jcansdale) [reply src/GitHub.App/Services/PullRequestService.cs]: Re: wondering if we could always use `new UTF8Encoding(false)`. Alas, this doesn't work. I tried the following:

```
File.WriteAllText("foo.txt", "foobar", new UTF8Encoding(true));
File.WriteAllText("bar.txt", "foobar", new UTF8Encoding(false));
```

When I diff them together, I see:
image

Comment (StanleyGoldman) [reply src/GitHub.App/Services/PullRequestService.cs]: This should change

**Pull request merged**